src/main/java/hudson/plugins/sonar/utils/SonarUtils.java

```diff
@@ -94,6 +94,20 @@ public static String extractSonarProjectURLFromLogs(Run<?, ?> build) throws IOEx
     }
     return filtered;
   }
+  
+  @CheckForNull
+  public static <T extends Action> T getPersistentAction(Actionable actionable, Class<T> type) {
+ // we use this method to avoid recursively calling transitive action factories
+    for (Action a : actionable.getActions()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "getActions"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)